lang/java/avro/src/main/java/org/apache/avro/Schema.java

```diff
@@ -565,6 +633,17 @@ public Field(String name, Schema schema) {
       this(name, schema, (String) null, (JsonNode) null, true, Order.ASCENDING);
     }
 
+    /**
+     * This is the data type of the field, e.g. if the field is defined as
+     * union[NULL, Long] the returned data type is AvroLong because the field should
+     * be set with Long values.
```

Comment:
This is also true for any UNION schema -- shouldn't `getField(0).getDataType()` here always be equivalent to `getField(0).schema().getDataType()`?